### DISCUSSION TITLE
Multiple definition of `histo_head' when installing NBIS Binaries

### DISCUSSION THREAD

**Discussion by Tim-Healey** [NoahGWood/OpenEFT#17] [Q&A] [answered]
I've gotten the program to startup just fine. On the last step, I am hit with an error it seems like some others are having.
![image](https://user-images.githubusercontent.com/1662817/229876775-cb8fcf2d-4794-4b22-bf6c-c3702a7ade84.png)

The NFSEG command is not installed when installing NBIS binaries. When executing `make it` during installation, I am met with this error message.
![image](https://user-images.githubusercontent.com/1662817/229877929-fcf6289c-1f49-4191-ada0-e70c615284a3.png)
I am using a fresh installation of Ubuntu 22.04.2 LTS.

Any help is appreciated.

**Comment by jstrahl3** [comment, marked as answer]
I was having the same issue and just figured it out tonight.  Apparently the newest versions of gcc aren't as forgiving to some of the poor coding of the past.  After much searching and trial and error I found that you need to install an old version of gcc to compile it.  After I did that everything worked as expected.  Below is the link with the instructions I used and the oldest version of gcc that I could install was 9.  Good luck!

https://stackoverflow.com/questions/63593100/how-to-install-older-version-of-gcc-3-or-gcc-4-in-ubuntu-20

**Reply by Tim-Healey** [reply]
Yep this was the problem. GCC-9 didn't work for me, but going to GCC-5 worked fine. 
This has the sources necessary to downgrade to 5. https://askubuntu.com/questions/1235819/ubuntu-20-04-gcc-version-lower-than-gcc-7

I am still getting nfseg non-zero exit status 254, but at least the command exists now.

**Reply by somekid6** [reply]
I'm having the same "multiple definition of 'histo_head'" issue when I get to the "make it" step. Tried gcc-5, gcc-7, gcc-9, and gcc-11 on Ubuntu 22.04.2 LTS but the issue persists.

**Reply by Tim-Healey** [reply]
@somekid6, maybe a dumb question but GCC 5 is the active version, right? I'm assuming you've added it to your path variable.
`gcc --version`
When you install the nbis binaries, what is the error message in terminal?

**Reply by somekid6** [reply]
@Tim-Healey Correct,  it's the active version. I think the path variable is correct. I'm using update-alternatives to swap between the versions.

![image](https://github.com/NoahGWood/OpenEFT/assets/53590807/b7bd4550-a25e-4e67-970b-c216e67899fb)


Just the error:

> Start: Compiling histogen binary....
make[4]: Entering directory '/home/kasse/OpenEFT/nbis/an2k/src/bin/histogen'
/usr/bin/gcc -m64 -fPIC /home/kasse/OpenEFT/nbis/an2k/obj/src/bin/histogen/histogen.o /home/kasse/OpenEFT/nbis/an2k/obj/src/bin/histogen/chkfile.o   -o /home/kasse/OpenEFT/nbis/an2k/bin/histogen
/usr/bin/ld: /home/kasse/OpenEFT/nbis/an2k/obj/src/bin/histogen/chkfile.o:(.bss+0x6c8): multiple definition of `histo_head'; /home/kasse/OpenEFT/nbis/an2k/obj/src/bin/histogen/histogen.o:(.bss+0x218): first defined here
collect2: error: ld returned 1 exit status
make[4]: *** [/home/kasse/OpenEFT/nbis/buildutil/bin.mak:68: /home/kasse/OpenEFT/nbis/an2k/bin/histogen] Error 1
make[4]: Leaving directory '/home/kasse/OpenEFT/nbis/an2k/src/bin/histogen'
make[3]: *** [/home/kasse/OpenEFT/nbis/buildutil/an2k_bins.mak:46: bins] Error 1
make[3]: Leaving directory '/home/kasse/OpenEFT/nbis/an2k/src/bin'
make[2]: *** [/home/kasse/OpenEFT/nbis/buildutil/src.mak:52: bins] Error 1
make[2]: Leaving directory '/home/kasse/OpenEFT/nbis/an2k/src'
make[1]: *** [/home/kasse/OpenEFT/nbis/buildutil/package.mak:211: bins] Error 1
make[1]: Leaving directory '/home/kasse/OpenEFT/nbis/an2k'
make: *** [Makefile:152: it] Error 1

Here's my full terminal output starting with gcc version and manually walking through the NBIS section of build_linux.sh. Error starts at line 1804.

[nbisAttempt.txt](https://github.com/NoahGWood/OpenEFT/files/11518065/nbisAttempt.txt)

**Reply by Tim-Healey** [reply]
Looks like someone has fixed the compilation error for newer versions of gcc. Try remove the directory and reinstall using this version of NBIS. https://github.com/OwenPemberton/nbis

**Reply by somekid6** [reply]
Sweet that got me past that error. Thanks a bunch! 

Deleted the nbis folder from this repo and cloned the [OwenPemberton/nbis](https://github.com/OwenPemberton/nbis) repo to OpenEFT/nbis so build_linux.sh still works.

edit: Spoke too soon. Apparently I did something in combination with this and it worked. Tested on a fresh Ubuntu install and the OwenPemberton repo is showing the same issue. Working on going through my notes to see what I did.

edit 2: I figured out what I broke. I switched back to the build_linux.sh script and the "git submodule update" was overwriting the new nbis.
